part-2.py

```diff
@@ -4,13 +4,55 @@
 # the appropriate comment.
 
 # search
-
+def search(array, query, i= 0):
+    if len(array) == i:
+        return False
+    elif array[i] == query:
+        return True
+    else:
+        return search(array, query, i + 1)
+# ALT METHOD
+# def search(array, query):
+#     if query == "" or array == []:
+#         return False
+#     if query not in array:
+#         return False
+#     if query in array:
+#         return True
+    
+#     return search(array, query)
+        
+#     if not len(array):
+#         return False
+#     elif array[0] == query:
+#         return True
+#     return search(array[1:], query)
 
 
 # is_palindrome
+def is_palindrome(text):
+    def helper(left, right):
+        if right - left <= 0:
+            return True
+        elif text[left] != text[right]:
+            return False
+        else:
+            return helper(left + 1, right - 1)
+            
+    return helper(0, len(text) - 1)
 
 
 
 # digit_match
-
-
+def digit_match(left, right, matches = 0):
+    if left == 0 and right == 0:
+        return matches + 1
+    elif left <10 or right < 10:
+        if left % 10 == right % 10:
+            return matches + 1
+        else:
+            return matches
+    elif left % 10 == right % 10:
```

Comment:
Note that `int(left % 10 == right % 10)` equal 1 or 0 if the conditional evaluates to true or false. Consider how you could use this to DRY up your code.